File: fix_layouts.py
#!/usr/bin/env python3
import sys

# Read original
with open('src/components/InteractiveTrain.tsx', 'r') as f:
    original = f.read()

lines = original.split('\n')

# Find where FACTORIES line is (should be around line 250-252)
factory_line = None
for i, line in enumerate(lines):
    if line.strip().startswith('const FACTORIES = ['):
        factory_line = i
        break

if factory_line is None:
    print("ERROR: Could not find FACTORIES line")
    sys.exit(1)

# Layouts end at factory_line (exclusive)
# Header is lines 0-13
# We replace lines 14 to factory_line with new content

print(f"FACTORIES found at line {factory_line+1}")
print(f"Header lines: 0-13 ({len(lines[:14])} lines)")
print(f"Layout replacement: lines 14-{factory_line-1} ({factory_line-14} lines)")
print(f"Keeping from line {factory_line} onwards")

# Read new layout content
with open('src/components/InteractiveTrain.layouts.txt', 'r') as f:
    new_layouts = f.read()

# Build new file
new_lines = lines[:14] + new_layouts.split('\n') + lines[factory_line:]
new_content = '\n'.join(new_lines)

with open('src/components/InteractiveTrain.tsx', 'w') as f:
    f.write(new_content)

print(f"Done! New file has {len(new_lines)} lines")

File: fix_layouts.js
const fs = require('fs');
const path = require('path');

const file = 'src/components/InteractiveTrain.tsx';
const content = fs.readFileSync(file, 'utf8');
const lines = content.split('\n');

// Find FACTORIES line
let factoryLine = -1;
for (let i = 0; i < lines.length; i++) {
  if (lines[i].trim().startsWith('const FACTORIES = [')) {
    factoryLine = i;
    break;
  }
}

if (factoryLine === -1) {
  console.error('FACTORIES not found');
  process.exit(1);
}

console.log(`FACTORIES at line ${factoryLine + 1}`);

// Header: lines 0-13
// Layouts: lines 14 to factoryLine-1
// Rest: factoryLine onwards

const newLayouts = fs.readFileSync('src/components/InteractiveTrain.layouts.txt', 'utf8');
const newLayoutLines = newLayoutLinesFn(newLayouts);

function newLayoutLinesFn(text) {
  return text.split('\n');
}

const newLines = [
  ...lines.slice(0, 14),
  ...newLayoutLines,
  ...lines.slice(factoryLine)
];

const newContent = newLines.join('\n');
fs.writeFileSync(file, newContent);
console.log(`Done! ${newLines.length} lines`);
